encoders/numeric/UMAPEncoder/tests/test_umapencoder.py

```diff
@@ -0,0 +1,55 @@
+import os
+import numpy as np
+import pytest
+
+from jina.executors import BaseExecutor
+from jina.executors.metas import get_default_metas
+
+from .. import UMAPEncoder
+
+
+target_output_dim = 2
+
+
+@pytest.fixture(scope="function", autouse=True)
+def metas(tmpdir):
+    metas = get_default_metas()
+    metas['workspace'] = str(tmpdir)
+    yield metas
+
+
+@pytest.fixture(scope="function")
+def test_data():
+    """
+    Create an array of the given shape and populate it with random samples from a uniform distribution over [0, 1).
+    :return: a `B x T` numpy ``ndarray``, `B` is the size of the batch
+    """
+    batch_size = 10
+    input_dim = 28
+    test_data = np.random.rand(batch_size, input_dim)
+
+    return test_data
+
+
+def get_encoder():
```

Comment:
maybe pass `target_output_dim` here so we can set it when using instead of the global setting.